app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -442,4 +468,77 @@ public void run() {
             },UNDELIVERED_PACKAGE_TIMEOUT);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * this functions converts a byte array into a file
+     * @param fileName name of the file with extension
+     * @param fileByte file bytes to convert
+     * @return a path to the newly written file
+     */
+    public String writeFileExternalStorage(String fileName, byte[] fileByte) {
+        String state = Environment.getExternalStorageState();
+        //external storage availability check
+        if (!Environment.MEDIA_MOUNTED.equals(state)) {
+            return null;
+        }
+        File file = new File(Environment.getExternalStoragePublicDirectory(
+                Environment.DIRECTORY_DOWNLOADS), fileName);
+        FileOutputStream outputStream = null;
+        try {
+            file.createNewFile();
+            //second argument of FileOutputStream constructor indicates whether to append or create new file if one exists
+            outputStream = new FileOutputStream(file, false);
+            outputStream.write(fileByte);
+            outputStream.flush();
+            outputStream.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
This should probably be communicated to the person that called this, so they can handle the failure appropriately.